tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -213,6 +213,11 @@ public static Tracer Instance
             {
                 lock (_globalInstanceLock)
                 {
+                    if (_instance is ILockedTracer)
+                    {
+                        throw new Exception("The current tracer instance cannot be replaced.");
```

Comment:
nit: I think throwing an `InvalidOperationException` would be even more descriptive

Reply:
👍🏻  done in 0d482c2